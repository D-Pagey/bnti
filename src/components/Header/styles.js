import styled from 'styled-components';
import { NavLink } from 'react-router-dom';
import { padding } from 'polished';
import { colour, fontSize, spacing } from '../../tokens';

export const Wrapper = styled.div`
    align-items: center;
    background-color: ${colour.blue};
    display: flex;
    justify-content: space-between;
    ${padding(spacing.medium)};
`;

export const Link = styled(NavLink)`
    color: inherit;
    font-size: ${fontSize.huge};
    text-decoration: none;
`;

export const Logo = styled.h2`
    margin: 0;
`;

export const White = styled.span`
    color: white;
`;
