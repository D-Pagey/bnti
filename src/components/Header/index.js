import React from 'react';
import Media from 'react-media';
import { mediaQuery } from '../../tokens';
import Navbar from '../Navbar';
import * as S from './styles';

const Header = () => {
    return (
        <S.Wrapper>
            <S.Logo>
                <S.Link to="/">
                    <S.White>BR</S.White>Battles
                </S.Link>
            </S.Logo>

            <Media query={mediaQuery.min.medium} render={() => <Navbar />} />
        </S.Wrapper>
    );
};

export default Header;
