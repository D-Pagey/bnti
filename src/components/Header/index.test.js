import React from 'react';
import { render } from 'react-testing-library';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import Header from '.';

const renderWithRouter = (ui) => render(<Router history={createMemoryHistory()}>{ui}</Router>);

describe('Header component', () => {
    it('should render', () => {
        const { container } = renderWithRouter(<Header />);
        expect(container.firstChild).toMatchSnapshot();
    });
});
