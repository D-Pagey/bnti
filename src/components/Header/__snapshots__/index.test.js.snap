// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render 1`] = `
.c4 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  list-style: none;
  margin: 0;
  padding: 0;
}

.c5 {
  margin-right: 2rem;
}

.c5:hover {
  color: white;
}

.c6 {
  color: inherit;
  font-size: 1.2rem;
  font-weight: 500;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c7 {
  background-color: black;
  border: 0;
  border-radius: 16px;
  color: white;
  cursor: pointer;
  font-size: 1.2rem;
  font-weight: 100;
  padding: 5px 15px;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #00A7CB;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 1.3rem 1rem;
}

.c2 {
  color: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c1 {
  margin: 0;
}

.c3 {
  color: white;
}

<div
  class="c0"
>
  <h2
    class="c1"
  >
    <a
      aria-current="page"
      class="c2 active"
      href="/"
    >
      <span
        class="c3"
      >
        BR
      </span>
      Battles
    </a>
  </h2>
  <ul
    class="c4"
  >
    <li
      class="c5"
    >
      <a
        class="c6"
        href="/about"
      >
        About
      </a>
    </li>
    <li
      class="c5"
    >
      <a
        class="c6"
        href="/tournaments"
      >
        Tournaments
      </a>
    </li>
    <li
      class="c5"
    >
      <a
        class="c6"
        href="/stats"
      >
        Stats
      </a>
    </li>
    <li
      class="c5"
    >
      <a
        class="c6"
        href="/login"
      >
        <button
          class="c7"
          type="button"
        >
          Login
        </button>
      </a>
    </li>
  </ul>
</div>
`;
